lib/notifiers/thomas2.dart

```diff
@@ -1,217 +1,90 @@
-import 'dart:collection';
-
 import 'package:flutter/foundation.dart';
 
-class _ListenerEntry extends LinkedListEntry<_ListenerEntry> {
-  VoidCallback listener;
+class _Listener {
+  _Listener(this.func);
+  final VoidCallback func;
+  bool active = true;
+  void call() {
+    if (active) {
+      func();
+    }
+  }
 }
 
-/// A class that can be extended or mixed in that provides a change notification
-/// API using [VoidCallback] for notifications.
-///
-/// It is O(1) for adding listeners and O(N) for removing listeners and dispatching
-/// notifications (where N is the number of listeners).
-///
-/// See also:
-///
-///  * [CustomLinkedListChangeNotifier], which is a [ChangeNotifier] that wraps a single value.
-class Thomas2ChangeNotifier implements Listenable {
-  LinkedList<_ListenerEntry> _listeners = LinkedList<_ListenerEntry>();
-  _ListenerEntry _notifyCursor;
-  _ListenerEntry _notifyLimit;
-
-  bool _debugAssertNotDisposed() {
-    assert(() {
-      if (_listeners == null) {
-        throw FlutterError('A $runtimeType was used after being disposed.\n'
-            'Once you have called dispose() on a $runtimeType, it can no longer be used.');
-      }
-      return true;
-    }());
-    return true;
-  }
+class Thomas2ChangeNotifier {
+  List<_Listener> _listeners = List<_Listener>();
+  int _notificationCallStackDepth = 0;
+  int _removedListeners = 0;
+  int _listenerLengh; // This is the length on the first not recursive call
```

Comment:
I think you wanted to wrote _listenerLength instead.

Reply:
yep